assets/js/modules/adsense/components/setup/v2/SetupMain.js

```diff
@@ -180,25 +131,15 @@ export default function SetupMain() {
 	// Update current account ID setting on-the-fly.
 	useEffect( () => {
 		if (
-			! accountID ||
-			accounts?.length !== 1 ||
-			previousAccountID ||
-			( accounts?.length === 1 &&
-				accounts.findIndex( ( { _id } ) => _id === accountID ) === -1 )
-		) {
-			return;
-		}
-
-		if (
-			( accounts?.length === 1 && ! previousAccountID ) ||
-			( accounts?.length === 1 &&
-				accounts.findIndex( ( { _id } ) => _id === accountID ) === -1 )
+			accounts?.length === 1 &&
+			( ! accountID ||
+				accounts.findIndex( ( { _id } ) => _id !== accountID ) ) === 0
```

Comment:
This is an overly complex condition and should be simplified (we know there's only 1 account as the test for `accounts?.length === 1` has already returned `true`):

```suggestion
				accounts[ 0 ]._id !== accountID
```

Reply:
Thanks @techanvil, that's a good catch 👍